lib/wasi/src/runtime/mod.rs

```diff
@@ -57,13 +59,33 @@ where
     }
 }
 
-#[derive(Clone, Debug)]
+#[derive(Debug, Default)]
+pub struct DefaultTty {
+    state: Mutex<WasiTtyState>,
+}
+
+impl TtyBridge for DefaultTty {
+    fn tty_get(&self) -> WasiTtyState {
+        let state = self.state.lock().unwrap();
+        state.clone()
+    }
+
+    fn tty_set(&self, tty_state: WasiTtyState) {
+        let mut state = self.state.lock().unwrap();
+        *state = tty_state;
+    }
+}
+
+#[derive(Clone, Derivative)]
+#[derivative(Debug)]
 pub struct PluggableRuntimeImplementation {
     pub rt: Arc<dyn VirtualTaskManager>,
     pub networking: DynVirtualNetworking,
     pub http_client: Option<DynHttpClient>,
     #[cfg(feature = "sys")]
     pub engine: Option<wasmer::Engine>,
+    #[derivative(Debug = "ignore")]
+    pub tty: Arc<dyn TtyBridge + Send + Sync>,
```

Comment:
Only WASIX environment want a TTY, WASI doesn't need it.

Reply:
I don't agree with this  as it outsources a code problem to a consumer, effectively the consumer has to determine if their binary will need TTY before executing it rather than the runtime handling that for them. The consequences of us supporting TTY by convention are trival on our side.

P.s. WASI did also need this for quite some time as I (along with others) had to create a hack for getting TTY in WASI by using the `/dev/tty` FD which has been upstreamed into create like the `password` crate which need to switch to none echo mode for password entering (quite common).

Reply:
I've made this a command line argument now however I default it to TTY attached rather than TTY detached (i.e. the command line argument is --no-tty)

Reply:
Oh, small mis-scommunication: I agree that the CLI and SSH can totally always enable the TTY. 

I was just saying it wouldn't be necessary for many WCGI or even many WASIX tasks that don't have interactivity by nature.